services/basic_types.proto

```diff
@@ -1493,3 +1493,67 @@ message TokenAssociation {
     TokenID token_id = 1;  // The token involved in the association
     AccountID account_id = 2; // The account involved in the association
 }
+
+/**
+ * An approved allowance for hbar transfers for a spender.
+ */
+message CryptoApproval {
+    /**
+     * The account ID of the spender of the hbar allowance.
+     */
+    AccountID spender = 1;
+
+    /**
+     * The amount of the spender's allowance in tinybars.
+     */
+    int64 amount = 2;
+}
+
+/**
+ * An approved allowance for fungible token transfers for a spender.
+ */
+message TokenApproval {
+    /**
+     * The token that the allowance pertains to.
+     */
+    TokenID tokenId = 1;
+
+    /**
+     * The account ID of the token allowance spender.
+     */
+    AccountID spender = 2;
+
+    /**
+     * The amount of the spender's token allowance.
+     */
+    int64 amount = 3;
+}
+
+/**
+ * An approved allowance for non-fungible token transfers for a spender.
+ */
+message NftApproval {
+    /**
+     * The token that the allowance pertains to.
+     */
+    TokenID tokenId = 1;
+
+    /**
+     * The account ID of the token allowance spender.
+     */
+    AccountID spender = 2;
+
+    oneof allowance {
+        /**
+         * The list of serial numbers that the spender is permitted to transfer.
+         */
+        repeated int64 serialNumbers = 3 [packed = true];
```

Comment:
`packed=true` is the default in proto3 and unnecessary to specify

Reply:
Fixed